mars/services/task/execution/ray/executor.py

```diff
@@ -250,17 +255,26 @@ def _get_subtask_out_info(
     # output_keys order should be consistent with remote `execute_subtask`,
     # dict can preserve insert order.
     output_keys = {}
+    shuffle_chunk = None
+    if is_mapper:
+        assert n_reducers is not None
+        if len(subtask_chunk_graph.result_chunks) == 1:
+            return set(), n_reducers
+        for chunk in subtask_chunk_graph.result_chunks:
+            if not chunk.is_mapper:
```

Comment:
I am not sure if the mapper result chunks should be checked the `isinstance(chunk.op, VirtualOperand)`

Reply:
`mapper result chunks` doesn't have `VirtualOperand` inside, it's safe to skip the check. Actually, I think it's safe to remove this check for normal chunks, but it's not related to this PR